### PR TITLE
vim-plugins: updates, add few plugins I use, colorschemes

### PR DESCRIPTION
**Pull request by dtzWill**
- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
The "rebuild" is because of the plugins -- nothing serious :).

File: pkgs/misc/vim-plugins/default.nix [hunk 1]
```diff
@@ -19,8 +19,9 @@ let
 
 
 # TL;DR
-# Add your plugin to ./vim-plugin-names
-# Regenerate via `nix-shell -I nixpkgs=/path/to/your/local/fork -p vimPlugins.pluginnames2nix --command "vim-plugin-names-to-nix +'w result' +qall"`
+# * Add your plugin to ./vim-plugin-names
+#   * sort -df ./vim-plugin-names > sorted && mv sorted vim-plugin-names
+# * Regenerate via `nix-shell -I nixpkgs=/path/to/your/local/fork -p vimPlugins.pluginnames2nix --command "vim-plugin-names-to-nix +silent +'x! result'"`
```
Comment (LnL7) [inline pkgs/misc/vim-plugins/default.nix]: Does that fix the pagination?

File: pkgs/misc/vim-plugins/default.nix [hunk 2]
```diff
@@ -192,11 +193,11 @@ let
   };
 
   vim-nix = buildVimPluginFrom2Nix { # created by nix#NixDerivation
-    name = "vim-nix-2018-04-24";
+    name = "vim-nix-2018-07-01";
```
Comment (LnL7) [inline pkgs/misc/vim-plugins/default.nix]: I made some changes to the indentation handling, let me know if you notice issues.

**Review by LnL7 — APPROVED** [review]

File: pkgs/misc/vim-plugins/default.nix (hunk 1 above)
Comment (dtzWill) [reply pkgs/misc/vim-plugins/default.nix]: The '+silent' did, for me anyway! :grin:

Comment (LnL7) [reply pkgs/misc/vim-plugins/default.nix]: Nice! We might be able to close https://github.com/NixOS/nixpkgs/issues/38748 then.
